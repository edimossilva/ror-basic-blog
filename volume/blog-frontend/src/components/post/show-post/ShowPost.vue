<template lang="">
  <div>
    <h3>Show Post</h3>

    <div>
      <p> Post Title: {{ post.title }} </p>
      <p> Post Owner: {{ post.owner }} </p>

    </div>

    <button @click="redirectToShowBlog">Show posts</button>
    <button @click="redirectToCreatePost">Create post</button>

  </div>
</template>

<script>
import postsApi from "../../../services/posts_api";

export default {
  props: {
    id: {
      required: true
    },
    blogId: {
      required: true
    }
  },

  data() {
    return {
      post: {}
    };
  },

  methods: {
    getPost() {
      const params = {
        id: this.id,
        blogId: this.blogId
      };
      postsApi
        .getPost({
          params,
          token: this.$store.getters.authToken
        })
        .then(this.updateData)
        .catch(this.showErrorMessage);
    },

    redirectToShowBlog() {
      this.$router.push({ name: "showBlog", params: { id: this.blogId } });
    },
    redirectToCreatePost() {
      this.$router.push({
        name: "createPost",
        params: { blogId: this.blogId }
      });
    },
    updateData({ data: { data } }) {
      this.post = data;
    },

    showErrorMessage({
      response: {
        data: { error_message }
      }
    }) {
      // debugger; // eslint-disable-line
      this.$modal.show("modal", { message: error_message });
    }
  },

  mounted: function() {
    this.getPost();
  }
};
</script>

<style scoped></style>