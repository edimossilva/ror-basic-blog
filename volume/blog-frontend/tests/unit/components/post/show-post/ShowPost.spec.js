import { shallowMount } from '@vue/test-utils'
import ShowPost from '@/components/post/show-post/ShowPost'

describe('ShowPost.vue', () => {
  it('is a valid component', () => {
    const mockPost = {
      owner: "mockOwner",
      title: "mockTitle",
    }

    const wrapper = shallowMount(ShowPost, {
      methods: { getPost: jest.fn() },
      propsData: {
        id: 1,
        blogId: 2
      },
      data() { return { post: mockPost } }
    });

    expect(wrapper.isVueInstance()).toBeTruthy()
    expect(wrapper).toMatchSnapshot()
  })
})
