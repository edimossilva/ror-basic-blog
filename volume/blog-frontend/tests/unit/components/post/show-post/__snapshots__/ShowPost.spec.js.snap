// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShowPost.vue is a valid component 1`] = `
<div>
  <h3>Show Post</h3>
  <div>
    <p> Post Title: mockTitle </p>
    <p> Post Owner: mockOwner </p>
  </div> <button>Show posts</button> <button>Create post</button>
</div>
`;
